depends/packages/boost.mk

```diff
@@ -1,7 +1,7 @@
 package=boost
 $(package)_version=1_70_0
 $(package)_download_path=https://dl.bintray.com/boostorg/release/1.70.0/source/
-$(package)_file_name=$(package)_$($(package)_version).tar.bz2
+$(package)_file_name=boost_$($(package)_version).tar.bz2
```

Comment:
Is this an unrelated improvement?

Reply:
It is related. I do think it's an improvement to not use our internal package name in a tarball URL that's generated and hosted by a third party and we don't control. But this change is actually needed to make the build work.

Specifically, this is needed to make native_boost package work, since it derives from the boost package but has a different package name. The native_boost package is a new package needed to build the native_libmultiprocess package which currently uses boost::optional. The native_libmultiprocess package is needed to run a code generation tool in cross-compiled builds (since a tool that runs at build time should be native, not cross-compiled). I'm probably going to try to drop the boost::optional dependency at some point, but for now just adding a native_boost package seemed like the path of least resistance, since boost::optional is so convenient.

Reply:
Now that `native_libmultiprocess` doesn't need `boost::optional`, do you still need this? If you do need it, maybe make that a separate commit.